From: Carlos Maddela <e7appew@gmail.com>
Date: Thu, 30 Mar 2017 04:41:15 +1100
Subject: Perform fstat on stdin and stdout only if required.

Description: Perform fstat on stdin and stdout only if required.
Author: Carlos Maddela <e7appew@gmail.com>
Bug-Debian: https://bugs.debian.org/831415
Forwarded: https://github.com/kfish/xsel/pull/22
Last-Update: 2017-03-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
---
 xsel.c | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/xsel.c b/xsel.c
index 32c4866..ccd239e 100644
--- a/xsel.c
+++ b/xsel.c
@@ -2132,18 +2132,22 @@ main(int argc, char *argv[])
     exit (0);
   }
 
-  if (fstat (0, &in_statbuf) == -1) {
-    exit_err ("fstat error on stdin");
-  }
-  if (fstat (1, &out_statbuf) == -1) {
-    exit_err ("fstat error on stdout");
+  if (do_input || force_input) {
+    if (fstat (0, &in_statbuf) == -1) {
+      exit_err ("fstat error on stdin");
+    }
+    if (S_ISDIR(in_statbuf.st_mode)) {
+      exit_err ("-: Is a directory\n");
+    }
   }
 
-  if (S_ISDIR(in_statbuf.st_mode)) {
-    exit_err ("-: Is a directory\n");
-  }
-  if (S_ISDIR(out_statbuf.st_mode)) {
-    exit_err ("stdout: Is a directory\n");
+  if (do_output || force_output) {
+    if (fstat (1, &out_statbuf) == -1) {
+      exit_err ("fstat error on stdout");
+    }
+    if (S_ISDIR(out_statbuf.st_mode)) {
+      exit_err ("stdout: Is a directory\n");
+    }
   }
 
   timeout = timeout_ms * 1000;
